### PR TITLE
Improve datagrid scrolling

### PR DESCRIPTION
**Pull request by maxkatz6**
## What does the pull request do?
1. Handles horizontal wheel scrolling in DataGrid
2. Handles touch scrolling in DataGrid using ScrollGestureRecognizer

## What is the current behavior?
scroll2

## Fixed issues
Fixes #https://github.com/AvaloniaUI/Avalonia/issues/3358

### CODE REVIEW

File: src/Avalonia.Controls.DataGrid/DataGridCell.cs [hunk 1]
```diff
@@ -173,7 +173,7 @@ private void DataGridCell_PointerPressed(PointerPressedEventArgs e)
                     }
                     if (OwningRow != null)
                     {
-                        e.Handled = OwningGrid.UpdateStateOnMouseLeftButtonDown(e, ColumnIndex, OwningRow.Slot, !e.Handled);
+                        OwningGrid.UpdateStateOnMouseLeftButtonDown(e, ColumnIndex, OwningRow.Slot, !e.Handled);
```
Comment (maxkatz6) [inline src/Avalonia.Controls.DataGrid/DataGridCell.cs:176]: Hard to say if it's breaking change or not.
If touch input is handled by cell selection, then it's not possible to scroll content.

Comment (jmacato) [reply src/Avalonia.Controls.DataGrid/DataGridCell.cs:176]: tis kinda weird to scroll inside cells anyway so probably not

Comment (maxkatz6) [reply src/Avalonia.Controls.DataGrid/DataGridCell.cs:176]: Well, in this case it's about "pointer pressing". Scroll gestures listen for this event to initiate a gesture.
But if this event was already handled somewhere, it doesn't scroll cell's parents (including data grid).

Comment (kekekeks) [reply src/Avalonia.Controls.DataGrid/DataGridCell.cs:176]: We might want to check the pointer type and keep the previous logic when mouse is used.

**Review by grokys — APPROVED** [review]

**Pull request merged**